### DISCUSSION TITLE
How to change 'examples/api-samples' to new name?

### DISCUSSION THREAD

**Discussion by saltylassi** [eclipse-theia/theia#12009] [Q&A] [answered]
Hello, I'm trying to change folder name to new folder name, but not working.

here is what I tried : 

1. clone theia
2. change folder name (api-samples => something-new-folder)
3. ```yarn install```

and got this errors : 
```
packages/editor-preview/src/browser/editor-preview-tree-decorator.ts:31:32 - error TS2307: Cannot find module '@theia/navigator/lib/browser/open-editors-widget/navigator-open-editors-tree-model' or its corresponding type declarations.

31 import { OpenEditorNode } from '@theia/navigator/lib/browser/open-editors-widget/navigator-open-editors-tree-model';
                                  ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

packages/editor-preview/src/browser/editor-preview-tree-decorator.ts:97:25 - error TS2339: Property 'widget' does not exist on type 'TreeNode'.

97                 const { widget } = node;
                           ~~~~~~

packages/editor-preview/src/browser/editor-preview-frontend-module.ts:26:42 - error TS2307: Cannot find module '@theia/navigator/lib/browser/open-editors-widget/navigator-open-editors-decorator-service' or its corresponding type declarations.

26 import { OpenEditorsTreeDecorator } from '@theia/navigator/lib/browser/open-editors-widget/navigator-open-editors-decorator-service';
           
```

I think It's a small modification, what did I do wrong?

**Comment by msujew** [comment, marked as answer]
Hey @saltylassi,

the project name is wired into a few more of our dev-tools, mostly TypeScript. That might be the reason why TypeScript starts complaining. Can you search the workspace for leftover traces of the `api-samples` name?

On another note, why do you want to change the name of the api-samples in the first place? If you're trying to create your own Theia extension, you should use the [yeoman generator for Theia](https://github.com/eclipse-theia/generator-theia-extension).

**Reply by saltylassi** [reply]
Thanks, I'll try yeoman generator. and this is update :


in some cases, I can build theia without error.

when I change folder name to ```a{word} or b{word}``` like ```arcReactor``` or ```blackPanther```, theia says "oh, that's nice. I like that."

If you're not busy, could you please check this case?



-----------------------------------------------------------------------------------------------------------------
nevermind. I think I find something

order of name(alphabetical) should be smaller than ```browser```  because of ```references``` in ```theia/tsconfig.json```.
